src/Components/WebAssembly/WebAssembly/src/Hosting/RootComponentMapping.cs

```diff
@@ -49,11 +50,29 @@ public RootComponentMapping([DynamicallyAccessedMembers(Component)] Type compone
         /// <param name="componentType">The component type. Must implement <see cref="IComponent"/>.</param>
         /// <param name="selector">The DOM element selector or registration id for the component.</param>
         /// <param name="parameters">The parameters to pass to the component.</param>
-        public RootComponentMapping([DynamicallyAccessedMembers(Component)] Type componentType, string selector, ParameterView parameters) : this(componentType, selector)
+        public RootComponentMapping([DynamicallyAccessedMembers(Component)] Type componentType, string selector, ParameterView parameters)
+            : this(componentType, selector)
         {
             Parameters = parameters;
         }
 
+        /// <summary>
+        /// Creates a new instance of <see cref="RootComponentMapping"/> with the provided <paramref name="componentType"/>
+        /// and <paramref name="selector"/>.
+        /// </summary>
+        /// <param name="componentType">The component type. Must implement <see cref="IComponent"/>.</param>
+        /// <param name="selector">The DOM element selector or registration id for the component.</param>
+        /// <param name="parameters">The parameters to pass to the component.</param>
+        /// <param name="appendContent">
+        /// If <c>true</c>, the child content of the root component will be appended to existing HTML content.
+        /// This is useful when treating the HTML <c>&lt;head&gt;</c> as a root component.
+        /// </param>
+        public RootComponentMapping([DynamicallyAccessedMembers(Component)] Type componentType, string selector, ParameterView parameters, bool appendContent)
```

Comment:
I know this is a bit of a curveball, so please just reject it if so, but would there be any value in *not* having an extra bool flag here, and instead establishing a convention on the selector? For example, `head::after`? The JS-side code could easily check if the selector ended with `::after` and use that.

The reason I'm bringing this up is that, if in the future we want to support other things like a `prependContent` alternative, then it will get really tough to have all these combinations of overloads, and not even clear how we could have an overload for prepend without clashing with the append one. Also it will mean you don't need all the extra plumbing for passing through the modifiers as the existing params will suffice:

I'm not saying we should support other pseudoselectors like `::before` now - the JS could just throw if it sees any pseudoselector other than `::after`.

Opinions?

Reply:
This seems like a great alternative. I'm happy to go ahead and make this change!